segfault on launch (v1.0.0-alpha.1)

**What steps did you take and what happened:**
Upgraded to velero v1.0.0-alpha.1 from v0.11.0 due to bug https://github.com/heptio/velero/issues/1283 which was fixed by https://github.com/heptio/velero/pull/1293


**What did you expect to happen:**
For velero to run successfully 

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

* `kubectl logs deployment/velero -n velero`
```
time="2019-04-24T19:53:49Z" level=info msg="setting log-level to INFO"
time="2019-04-24T19:53:49Z" level=info msg="Starting Velero server v1.0.0-alpha.1 (39bab5ada91f1ed6fc03006a9f1d73dd356b28a5)" logSource="pkg/cmd/server/server.go:165"
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pod
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pv
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/serviceaccount
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=VolumeSnapshotter logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/aws
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=VolumeSnapshotter logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/azure
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=VolumeSnapshotter logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/gcp
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=ObjectStore logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/aws
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=ObjectStore logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/azure
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=ObjectStore logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/gcp
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/addPVCFromPod
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/addPVFromPVC
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/job
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pod
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/restic
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service
time="2019-04-24T19:53:49Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/serviceaccount
time="2019-04-24T19:53:49Z" level=info msg="Checking existence of namespace" logSource="pkg/cmd/server/server.go:355" namespace=velero
time="2019-04-24T19:53:49Z" level=info msg="Namespace exists" logSource="pkg/cmd/server/server.go:361" namespace=velero
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xd8 pc=0x10a045c]

goroutine 1 [running]:
github.com/heptio/velero/pkg/discovery.refreshServerPreferredResources(0x7f2aa54073c8, 0xc00040b920, 0x0, 0x0, 0xc00040b920, 0x0, 0x0, 0xc000044070, 0xc000044000)
	/go/src/github.com/heptio/velero/pkg/discovery/helper.go:153 +0x11c
github.com/heptio/velero/pkg/discovery.(*helper).Refresh(0xc0005a1680, 0x0, 0x0)
	/go/src/github.com/heptio/velero/pkg/discovery/helper.go:114 +0x38d
github.com/heptio/velero/pkg/discovery.NewHelper(0x2344040, 0xc00040b920, 0x2366260, 0xc0000be640, 0x0, 0x0, 0xc000675a88, 0xc000675a90)
	/go/src/github.com/heptio/velero/pkg/discovery/helper.go:76 +0x5e
github.com/heptio/velero/pkg/cmd/server.(*server).initDiscoveryHelper(0xc0005449c0, 0xc0000a9118, 0x6)
	/go/src/github.com/heptio/velero/pkg/cmd/server/server.go:368 +0x54
github.com/heptio/velero/pkg/cmd/server.(*server).run(0xc0005449c0, 0x0, 0x0)
	/go/src/github.com/heptio/velero/pkg/cmd/server/server.go:324 +0xf8
github.com/heptio/velero/pkg/cmd/server.NewCommand.func1(0xc000523b80, 0x36e7f28, 0x0, 0x0)
	/go/src/github.com/heptio/velero/pkg/cmd/server/server.go:188 +0x5b0
github.com/heptio/velero/vendor/github.com/spf13/cobra.(*Command).execute(0xc000523b80, 0x36e7f28, 0x0, 0x0, 0xc000523b80, 0x36e7f28)
	/go/src/github.com/heptio/velero/vendor/github.com/spf13/cobra/command.go:766 +0x2ae
github.com/heptio/velero/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0000e6f00, 0x6, 0xc0000e6f00, 0x6)
	/go/src/github.com/heptio/velero/vendor/github.com/spf13/cobra/command.go:852 +0x2ec
github.com/heptio/velero/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/github.com/heptio/velero/vendor/github.com/spf13/cobra/command.go:800
main.main()
	/go/src/github.com/heptio/velero/cmd/velero/main.go:34 +0x8a
```

**Anything else you would like to add:**


**Environment:**

- Velero version (use `velero version`): 
v1.0.0-alpha.1
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.2", GitCommit:"81753b10df112992bf51bbc2c2f85208aad78335", GitTreeState:"clean", BuildDate:"2018-05-12T04:12:47Z", GoVersion:"go1.9.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.5", GitCommit:"753b2dbc622f5cc417845f0ff8a77f539a4213ea", GitTreeState:"clean", BuildDate:"2018-11-26T14:31:35Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}

- Kubernetes installer & version:
AKS

- Cloud provider or hardware configuration:
AKS

- OS (e.g. from `/etc/os-release`):
AKS

--- a/pkg/discovery/helper.go
+++ b/pkg/discovery/helper.go
@@ -72,6 +72,7 @@ var _ Helper = &helper{}
 func NewHelper(discoveryClient discovery.DiscoveryInterface, logger logrus.FieldLogger) (Helper, error) {
 	h := &helper{
 		discoveryClient: discoveryClient,
+		logger:          logger,
 	}
 	if err := h.Refresh(); err != nil {
 		return nil, err

